lib/PuppeteerSharp/Frame.cs

```diff
@@ -178,36 +178,22 @@ internal Frame(CDPSession client, Page page, Frame parentFrame, string frameId)
         /// <param name="selector">A selector of an element to wait for</param>
         /// <param name="options">Optional waiting parameters</param>
         /// <returns>A task that resolves when element specified by selector string is added to DOM</returns>
+        /// <seealso cref="WaitForXPathAsync(string, WaitForSelectorOptions)"/>
         /// <seealso cref="Page.WaitForSelectorAsync(string, WaitForSelectorOptions)"/>
-        public async Task<ElementHandle> WaitForSelectorAsync(string selector, WaitForSelectorOptions options = null)
-        {
-            options = options ?? new WaitForSelectorOptions();
-            const string predicate = @"
-              function predicate(selector, waitForVisible, waitForHidden) {
-              const node = document.querySelector(selector);
-              if (!node)
-                return waitForHidden;
-              if (!waitForVisible && !waitForHidden)
-                return node;
-              const style = window.getComputedStyle(node);
-              const isVisible = style && style.visibility !== 'hidden' && hasVisibleBoundingBox();
-              const success = (waitForVisible === isVisible || waitForHidden === !isVisible);
-              return success ? node : null;
-
-              function hasVisibleBoundingBox() {
-                const rect = node.getBoundingClientRect();
-                return !!(rect.top || rect.bottom || rect.width || rect.height);
-              }
-            }";
-            var polling = options.Visible || options.Hidden ? WaitForFunctionPollingOption.Raf : WaitForFunctionPollingOption.Mutation;
-            var handle = await WaitForFunctionAsync(predicate, new WaitForFunctionOptions
-            {
-                Timeout = options.Timeout,
-                Polling = polling
-            }, selector, options.Visible, options.Hidden);
-            return handle as ElementHandle;
-        }
+        public Task<ElementHandle> WaitForSelectorAsync(string selector, WaitForSelectorOptions options = null)
+            => WaitForSelectorOrXPathAsync(selector, false, options);
 
+        /// <summary>
+        /// Waits for a selector to be added to the DOM
+        /// </summary>
+        /// <param name="xpath">A xpath selector of an element to wait for</param>
+        /// <param name="options">Optional waiting parameters</param>
+        /// <returns>A task that resolves when element specified by selector string is added to DOM</returns>
+        /// <seealso cref="WaitForSelectorAsync(string, WaitForSelectorOptions)"/>
+        /// <seealso cref="Page.WaitForXPathAsync(string, WaitForSelectorOptions)"/>
+        public Task<ElementHandle> WaitForXPathAsync(string xpath, WaitForSelectorOptions options = null)
```

Comment:
Do you want to add a code example like this? https://github.com/kblok/puppeteer/commit/cb684ebbc4de271c7f2187f4e52c0b3a83cc0122#diff-0f426376b964e609c5c4618eb8bed3b2R1299

Reply:
Cde example from here https://github.com/GoogleChrome/puppeteer/blob/v1.5.0/docs/api.md#framewaitforxpathxpath-options @Meir017 ?